### DISCUSSION TITLE
Click and drag to create new items

### DISCUSSION THREAD

**Discussion by Innders** [samuelarbibe/dnd-timeline#63] [Ideas] [answered]
I love the library, thank you very much for creating it 🙏 

What would be your suggestion on how to create a feature where you can click on a row and drag to create a new item.

My initial idea would to:

1. `onPointerDown` event creates a new item with `start` and `end` as the same.
2. track mouse position and update `end` to match.

Second idea:

Handle it completely outside of dnd-timeline and then just create the new item.

_I can imagine this is going to be a pretty popular request though!_

## **Update**

Clicking and dragging a new temporary item is relatively easy using `onPointerMove`. The biggest challenge I face now is converting relative pixel values into absolute range values. 

I assume `getValueFromScreenX` and `pixelsToValue` is what I'm looking for. [userTimelineContext Docs](https://samuel-arbibe.gitbook.io/dnd-timeline/documentation/timelinecontext/usetimelinecontext)

https://github.com/user-attachments/assets/249e6260-9ef7-4144-bb38-301a09c8e330

**Comment by samuelarbibe** [comment]
Hey!
I actually have already implemented this exact behavior in an internal project I work on.

The solution was to create a custom component that is loaded when a `mousedown` event is registered on a row. When loaded it should also receive the initial mouse position.

When loaded, this component registers a `mousemove` listener, and changes the internal width of the custom component by calculating the delta using `getRangeFromPixels`.

When a `mouseup` event is triggered, the custom component calls a callback with the final range, and then you can create your item like usual.

If I have spare time I will gladly create a simple example and add it to the existing examples.

Good luck 🍀

**Reply by Innders** [reply]
I have ended up with almost the exact same implementation. I'm just trying to figure out how to do the pixel to ranger conversion now.

Is `getRangeFromPixels` a function you have written because I don't see it in the library?

An example project with this I think would be very useful. Or even just a code snippet here! 💚 

Thanks!

**Comment by samuelarbibe** [comment, marked as answer]
Hey!  
A little update: I have created a new example that does exactly what you described.  
I have also added a new function to the library that I thought would be useful, so that a bonus.  
  
Here is an example of the behavior: https://dnd-timeline-create.vercel.app/
And here is a PR: #65.
 Would love you to go over the changes and approve them. You can look over `CreateItem.tsx`, all the rest has been copied from the `basic` example so no need to go over it.

**Reply by Innders** [reply]
Wow this is awesome! I managed to figure out my own method so I'm interested to see how it compares. Thanks 🙏
